modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperEngine.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.speechtotext.impl.engine;
+
+import org.opencastproject.speechtotext.api.SpeechToTextEngine;
+import org.opencastproject.speechtotext.api.SpeechToTextEngineException;
+import org.opencastproject.util.IoSupport;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+/** Whisper implementation of the Speech-to-text engine interface. */
+@Component(
+    immediate = true,
+    service = {
+        SpeechToTextEngine.class
+    },
+    property = {
+        "service.description=Whisper implementation of the SpeechToTextEngine interface",
+        "service.pid=org.opencastproject.speechtotext.impl.engine.WhisperEngine",
+        "engineType=whisper"
+    }
+)
+
+public class WhisperEngine implements SpeechToTextEngine {
+
+  private static final Logger logger = LoggerFactory.getLogger(WhisperEngine.class);
+
+  /** Name of the engine. */
+  private static final String engineName = "Whisper";
+
+  /** Config key for setting the path to Whisper. */
+  private static final String WHISPER_EXECUTABLE_PATH_CONFIG_KEY = "whisper.root.engine.path";
```

Comment:
True for whisper, but I left configurable because there are several whisper forks right now to try,  One of the most interesting is [Whisper's code translation to C++](https://github.com/ggerganov/whisper.cpp)